bin/tag_release.nim

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env -S nim r --verbosity:0 --skipParentCfg:on
+
+## Running this file will:
+## 1. Switch to the `main` branch, if there are no uncommitted changes (exiting
+##    otherwise).
+## 2. Pull the upstream `main` branch from github.com/exercism/configlet.
+## 3. Check that the local `main` branch is not ahead of upstream.
+## 4. Check that the most recent commit on the `main` branch is an untagged
+##    release commit.
+## 5. Prompt the user to tag the commit and push the tag to exercism/configlet,
+##    which will trigger a build job and create a draft release.
+import std/[strformat, strscans, strutils]
+import "."/bump_version
+
+proc checkRepoIsInPreTagState: string =
```

Comment:
I'm not entirely sure about the name, as the proc doesn't only check the state, but actually return the tag version.